store/localstore/local_region.go

```diff
@@ -44,18 +46,133 @@ type regionResponse struct {
 
 const chunkSize = 64
 
+type sortRow struct {
+	key  []types.Datum
+	meta tipb.RowMeta
+	data []byte
+}
+
+type topnContainer struct {
+	orderByItems []*tipb.ByItem
+	rows         []*sortRow
+	err          error
+}
+
+func (t *topnContainer) Len() int {
+	return len(t.rows)
+}
+
+func (t *topnContainer) Swap(i, j int) {
+	t.rows[i], t.rows[j] = t.rows[j], t.rows[i]
+}
+
+func (t *topnContainer) Less(i, j int) bool {
+	for index, by := range t.orderByItems {
+		v1 := t.rows[i].key[index]
+		v2 := t.rows[j].key[index]
+
+		ret, err := v1.CompareDatum(v2)
+		if err != nil {
+			t.err = errors.Trace(err)
+			return true
+		}
+
+		if by.Desc {
+			ret = -ret
+		}
+
+		if ret < 0 {
+			return true
+		} else if ret > 0 {
+			return false
+		}
+	}
+
+	return false
+}
+
+type topnSolver struct {
```

Comment:
topnContainer implements the `sort.Interface`, why not name it more explicitly, like `topnSorter`? And topnSolver implements `Heap` interface, why not name it `topnHeap`.

Why embed topnContainer in topnSolver? 
It's convenient to write, but one doesn't belong to another conceptually.